src/e20eb0.c

```diff
@@ -4,8 +4,157 @@
+
+        sp20.raster = D_802B7710_E215C0;
+        sp20.palette = D_802B7910_E217C0;
+        sp20.width = D_802B7710_E215C0_width;
+        sp20.height = D_802B7710_E215C0_height;
+        sp20.xOffset = -16;
+        sp20.yOffset = 26;
+        sp20.opacity = 255;
+        fold_update(0, FOLD_TYPE_NONE, 0, 0, 0, 0, 0x440);
+        fold_update(0, FOLD_TYPE_6,
+                    D_802B79A8_E21858->unk_2B, D_802B79A8_E21858->unk_2B, D_802B79A8_E21858->unk_2B, 0xFF, 0x440);
+        fold_appendGfx_component(0, &sp20, 0x40, sp78);
+
+        gSPPopMatrix(gMasterGfxPos++, G_MTX_MODELVIEW);
+    }
+}
+
+void func_802B742C(void) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    s32 var_a2 = 255;
+    f32 unk10;
+    Npc* npc;
+
+    if (((playerStatus->flags &
+            (PLAYER_STATUS_FLAGS_HAS_CONVERSATION_NPC |
+             PLAYER_STATUS_FLAGS_40000 |
+             PLAYER_STATUS_FLAGS_20)) != PLAYER_STATUS_FLAGS_HAS_CONVERSATION_NPC) ||
+        (gGameState == 4) ||
+        (playerStatus->animFlags & PLAYER_STATUS_ANIM_FLAGS_HOLDING_WATT) ||
+        (playerStatus->inputEnabledCounter != 0) ||
+        (playerStatus->animFlags & PLAYER_STATUS_ANIM_FLAGS_SPINNING))
+    {
+        D_802B79A8_E21858->unk_2A = 3;
+    }
+
+    switch (D_802B79A8_E21858->unk_2A) {
+        case 0:
+            D_802B79A8_E21858->unk_10 = 0.0f;
+            D_802B79A8_E21858->unk_14 += 0.05;
+            if (D_802B79A8_E21858->unk_14 >= 0.55) {
+                D_802B79A8_E21858->unk_14 = 0.55f;
+                D_802B79A8_E21858->unk_2A++;
+                D_802B79A8_E21858->unk_20 = 12;
+            }
+            break;
+        case 1:
+            D_802B79A8_E21858->unk_10 = 0.0f;
+            D_802B79A8_E21858->unk_20--;
+            if (D_802B79A8_E21858->unk_20 <= 0) {
+                D_802B79A8_E21858->unk_2A++;
+            }
+            break;
+        case 2:
+            unk10 = D_802B79A8_E21858->unk_10;
+            unk10 += 10.0f;
+            if (unk10 >= 360.0f) {
+                unk10 -= 360.0f;
+            }
+            D_802B79A8_E21858->unk_10 = unk10;
+            var_a2 = e20eb0_UnkAngleFunc1(unk10);
+            break;
+        case 3:
+            unk10 = D_802B79A8_E21858->unk_10;
+            unk10 += 25.0f;
+            if (unk10 >= 360.0f) {
+                unk10 -= 360.0f;
+            }
+            D_802B79A8_E21858->unk_10 = unk10;
+            if ((unk10 >= 70.0f && unk10 <= 110.0f) || (unk10 >= 250.0f && unk10 <= 290.0f)) {
+                D_8010C940 = NULL;
+                playerStatus->encounteredNPC = NULL;
+                playerStatus->animFlags &= ~PLAYER_STATUS_ANIM_FLAGS_SPEECH_PROMPT_AVAILABLE;
+                return;
+            }
+            var_a2 = e20eb0_UnkAngleFunc1(unk10);
+            break;
+    }
+
+    D_802B79A8_E21858->unk_2B = var_a2;
+    if (playerStatus->encounteredNPC != NULL) {
+        if (playerStatus->encounteredNPC != D_802B79A8_E21858->unk_00) {
```

Comment:
Can type this D_xxx as Npc*
if you dont want to do the replace maybe throw a comment on it